### PR TITLE
docs(storybook): upgrading using storybook cli

### PR DESCRIPTION
**Pull request by mandarini**
A guide to explain how to use `upgrade` and `automigrate` from Storybook CLI (essentially explain how to use the `config-dir` flag that Storybook offers)

### CODE REVIEW

**Review by isaacplmann — APPROVED** [review]
This guide looks good, but there should be a section describing how you should use the three update commands together.

When should you use `nx migrate`? When should you use `storybook upgrade`?  When should you use `storybook automigrate`?  Should ever use them all together?

What is the recommended set of commands if I just want to keep my repository up to date?

**Comment by mandarini** [comment]
Ok, I'll update this!! Thanks Isaac!

File: docs/shared/recipes/storybook/upgrading-storybook.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+## When should you use the Storybook CLI to upgrade?
+
+Nx offers migration generators to help you upgrade your Nx workspace and your projects to the latest version of Nx, and of the tools Nx uses. One of these tools is, of course, Storybook, and Nx offers migration generators to help you upgrade your Storybook setup as well. However, the Storybook CLI provides a more comprehensive set of commands and generators (_"automigrations"_ as they are called in their context) to help you upgrade your Storybook setup. It can be used in conjunction with the Nx migration generators to ensure your Storybook setup is up-to-date with the latest Storybook features and best practices.
+
+So, you should prefer to use the Storybook CLI to upgrade your Storybook setup in the following cases:
+
+- When you want to try out a new version of Storybook, for example the `next` version, for which Nx might not have a migration generator yet.
+
+- When Nx may does not already have a migrator for a specific Storybook version in place. Nx will in most cases take care of migrations that are specific to Nx, but the Storybook CLI automigrations are more comprehensive.
```
Comment (isaacplmann) [inline docs/shared/recipes/storybook/upgrading-storybook.md:18]:
```suggestion
- When Nx does not already have a migrator for a specific Storybook version in place. Nx will in most cases take care of migrations that are specific to Nx, but the Storybook CLI automigrations are more comprehensive.
```

**Review by isaacplmann — APPROVED** [review]
One small fix.   Approved so you can merge when you're ready

**Pull request merged**